src/main/java/org/graylog/integrations/ipfix/codecs/IpfixCodec.java

```diff
@@ -82,18 +88,52 @@
     protected IpfixCodec(@Assisted Configuration configuration, IpfixAggregator ipfixAggregator) throws IOException {
         super(configuration);
         this.ipfixAggregator = ipfixAggregator;
-        // Standard IPFIX definition file is loaded from the classpath and is mandatory. We don't check for null or empty.
+
         final URL standardIPFixDefTemplate = Resources.getResource(IpfixCodec.class, IPFIX_STANDARD_DEFINITION);
-        final String ipFixCustomDefPath = (String) configuration.getSource().get(CK_IPFIX_DEFINITION_PATH);
-        if (ipFixCustomDefPath == null || ipFixCustomDefPath.trim().isEmpty()) {
+        final List<String> customDefFilePathList = configuration.getList(CK_IPFIX_DEFINITION_PATH);
+        final List<URL> filePaths = new ArrayList<>();
+
+        if (customDefFilePathList == null || customDefFilePathList.isEmpty()) {
             infoElementDefs = new InformationElementDefinitions(standardIPFixDefTemplate);
         } else {
-            final URL customIPFixDefURL = Paths.get(ipFixCustomDefPath).toUri().toURL();
-            infoElementDefs = new InformationElementDefinitions(standardIPFixDefTemplate, customIPFixDefURL);
+            checkValidFilePath(customDefFilePathList);
+            filePaths.add(standardIPFixDefTemplate);
+            for (String filePath : customDefFilePathList) {
+                URL customDefURL = url(filePath.trim());
+                filePaths.add(customDefURL);
+            }
+            URL[] urls = convertToArray(filePaths);
+            infoElementDefs = new InformationElementDefinitions(urls);
         }
         this.parser = new IpfixParser(this.infoElementDefs);
     }
 
+    URL url(String s) throws MalformedURLException {
+        return Paths.get(s).toUri().toURL();
+    }
+
+    URL[] convertToArray(List<URL> urls) {
+        URL[] urlArray = new URL[urls.size()];
+        return urls.toArray(urlArray);
+    }
+
+    void checkValidFilePath(List<String> customDefFilePathList) throws IpfixException {
+        for (String filePath : customDefFilePathList) {
+            File file = new File(filePath.trim());
+            validateFilePath(file);
+        }
+    }
+
+    public void validateFilePath(File customDefFile) throws IpfixException {
+        if (customDefFile.isDirectory()) {
+            String message = "The specified path is a folder. Please specify the full path to the file.";
```

Comment:
The `message` variable is not needed here. The string message can be explicitly provided within the `new IpfixException(message)`